### ISSUE TITLE
Add $path negation for cosmetic rules

### ISSUE THREAD

**Issue by sfionov** [AdguardTeam/CoreLibs#1725]
Now it is not possible to easily create negated rules (you should create an regexp)

It will be nice to add some syntax for path negation

We may also add support for uBO syntax `:not(:matches-path())`.

**Comment by sfionov** [comment]
It will be simpler to convert `:not(:matches-path())` rules if we support path negation. Especially if it already contains a regexp.

**Comment by ameshkov** [comment]
I'd just like to not add more modifiers if it can be avoided. Even a regular expression is easily converted: `regex -> (!?regex)`

**Comment by stanislav-atr** [comment]
This was already done as part of [this issue](https://github.com/AdguardTeam/tsurlfilter/issues/30).

**Comment by ameshkov** [comment]
@stanislav-atr how exactly was the conversion done? Negative lookhead?

**Comment by stanislav-atr** [comment]
yes, for example, rule `ya.ru##:not(:matches-path(/\/(sub1|sub2)\/page\.html/))p` is being converted into `[$path=/^((?!\\/(sub1|sub2)\\/page\\.html).)*$/]ya.ru##p`

**Comment by stanislav-atr** [comment]
@ameshkov https://github.com/AdguardTeam/tsurlfilter/blob/master/packages/tsurlfilter/src/rules/rule-converter.ts#L595
